### PR TITLE
Throw exception if invalid filename is provided

### PR DESCRIPTION
**Pull request by datagutten**
Throw an exception if an invalid path is provided in the filename option.
Before this change, the script will continue and try to write and close a file pointer which does not exist.

### CODE REVIEW

**Comment by jrfnl** [comment]
@datagutten  Hiya! Sorry for the long wait, but we're finally trying to get through the list of open PRs.

We're currently planning two triage sessions for Requests 2.0 and would like to invite you to join us in one or both of these sessions to talk us through the PR and discuss it.

The triage sessions are currently planned for:
* Friday August 20, 07:00 - 11:00 UTC
* Friday September 3, 07:00 - 11:00 UTC

Would you be available to join us during those times on one of those days ?

File: library/Requests/Transport/cURL.php [hunk 1]
```diff
@@ -135,7 +135,11 @@ public function request($url, $headers = array(), $data = array(), $options = ar
 		$options['hooks']->dispatch('curl.before_send', array(&$this->handle));
 
 		if ($options['filename'] !== false) {
-			$this->stream_handle = fopen($options['filename'], 'wb');
+			@$this->stream_handle = fopen($options['filename'], 'wb');
```
Comment (schlessera) [inline library/Requests/Transport/cURL.php]: The error silencing should be as limited as possible, so we want ot have it directly in front of the problematic `fopen()` call.
```suggestion
			$this->stream_handle = @fopen($options['filename'], 'wb');
```

File: library/Requests/Transport/fsockopen.php [hunk 2]
```diff
@@ -234,7 +234,11 @@ public function request($url, $headers = array(), $data = array(), $options = ar
 		$doingbody = false;
 		$download = false;
 		if ($options['filename']) {
-			$download = fopen($options['filename'], 'wb');
+			@$download = fopen($options['filename'], 'wb');
```
Comment (schlessera) [inline library/Requests/Transport/fsockopen.php]:
```suggestion
			$download = @fopen($options['filename'], 'wb');
```

**Review by schlessera — CHANGES_REQUESTED** [review]

File: library/Requests/Transport/fsockopen.php [hunk 3]
```diff
@@ -234,7 +234,11 @@ public function request($url, $headers = array(), $data = array(), $options = ar
...
+			if($download===false) {
```
Comment (schlessera) [inline library/Requests/Transport/fsockopen.php]:
```suggestion
			if($download === false) {
```

**Review by schlessera — CHANGES_REQUESTED** [review]

**Comment by jrfnl** [comment]
* Rebased the PR to make it mergable.
* Added an extra commit with minor tweaks after review.

**Comment by jrfnl** [comment]
Added a second commit with minor tweaks to the test and adjustment to an existing test to allow for the exception. This should now be good to go.

**Review by schlessera — APPROVED** [review]

**Pull request merged**